packages/melody-hooks/__tests__/ComponentSpec.js

```diff
@@ -600,4 +600,25 @@ describe('component', () => {
         assert.equal(root.outerHTML, '<div><div>test</div></div>');
         assert.equal(mounted, 1);
     });
+    it('should recover from errors in the component function', () => {
+        const template = {
+            render(_context) {
+                elementOpen('div', null, null);
+                text(_context.value);
+                elementClose('div');
+            },
+        };
+
+        const root = document.createElement('div');
+        const MyComponent = createComponent(props => {
+            return { value: props.value };
```

Comment:
Should we maybe throw explicitly? Took me a while to spot why this would throw.

Reply:
Yes probably a good idea, will fix it